src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -97,7 +100,14 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 var result = await binder.BindModelAsync(bindingContext);
                 if (result != null)
                 {
-                    return result;
+                    if (result.IsModelSet || bindingContext.ModelState.ContainsKey(bindingContext.ModelName))
```

Comment:
sure. I'll add a comment.

added this code a while ago but believe the reason was conversion errors.  if a binder found _any_ value, we don't want to fall back to the empty prefix.  could also change `TypeConverterModelBinder` to return a result with `IsModelSet == true` when it hits a conversion `Exception`. but the effect would be the same and might lead to other behaviour changes elsewhere. worth trying from your POV?